### PR TITLE
[E2E] Add Graphs case

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**

### CODE REVIEW

**Review by TejaX-Alaghari — APPROVED** [review]
LGTM

**Review by the-slow-one — APPROVED** [review]

File: features/feature_case/graph/graph.cu [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
```
Comment (yihwang-nv) [inline features/feature_case/graph/graph.cu:9]: I think we should add a license file header.

File: features/feature_case/graph/graph.cu [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  cudaGraphInstantiate(&execGraph, graph, NULL, NULL, 0);
+
+  cudaGraphLaunch(execGraph, stream);
+
+  cudaStreamSynchronize(stream); // Ensure the graph has completed execution
+
+  cudaMemcpy(h_a, d_a, 10 * sizeof(float), cudaMemcpyDeviceToHost);
+
+  for (int i = 0; i < 10; i++) {
+    if (h_a[i] != 2.0f) {
```
Comment (yihwang-nv) [inline features/feature_case/graph/graph.cu:57]: For a floating point equality, I think we should use the form like `fabs(h_a[i] - 2.0) < 1e-6`.

Comment (daiyaan-ahmed6) [reply features/feature_case/graph/graph.cu:57]: Done

File: features/feature_case/graph/graph.cu [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// ====------ graph.cu ------------------------------------ *- CUDA -* ---===//
```
Comment (yihwang-nv) [inline features/feature_case/graph/graph.cu]:
```suggestion
// ===------- graph.cu ------------------------------------ *- CUDA -* ---===//
```

**Review by ShengchenJ — APPROVED** [review]

File: features/feature_case/graph/graph.cu (hunk 3 above)
Comment (daiyaan-ahmed6) [reply features/feature_case/graph/graph.cu]: Done

**Pull request merged**